Realm/RLMRealm.mm

```diff
@@ -272,12 +272,18 @@ + (RLMAsyncOpenTask *)asyncOpenWithConfiguration:(RLMRealmConfiguration *)config
                 }
                 return;
             }
-            dispatch_async(callbackQueue, ^{
+            // Ensure that we keep the Realm open until we've initialized the
+            // RLMRealm on the target queue. This ensures that the existing
+            // RealmCoordinator and DB are reused rather than being reopened.
+            // We can't just capture the ThreadSafeReference as dispatch_async()
+            // requires a copyable block.
+            dispatch_async(callbackQueue, [=, ref = ref.resolve<std::shared_ptr<Realm>>(nullptr)]() mutable {
```

Comment:
Why are we resolving with a nullptr payload?

Reply:
The argument there is the realm to resolve the TSR in, which isn't applicable when resolving a TSR to a realm.